File: 2316-count-hills-and-valleys-in-an-array/count-hills-and-valleys-in-an-array.py
class Solution:
    def countHillValley(self, nums: List[int]) -> int:
        n=[nums[0]]
        for num in nums[1:]:
            if num != n[-1]:
                n.append(num)
        hills=0
        for i in range(1,len(n)-1):
            if n[i]>n[i-1] and n[i]>n[i+1]:
                hills+=1
            elif n[i]<n[i-1] and n[i]<n[i+1]:
                hills+=1
        return hills

File: 2316-count-hills-and-valleys-in-an-array/count-hills-and-valleys-in-an-array.cs
public class Solution {
    public int CountHillValley(int[] nums) {
        List<int> n = new List<int> {nums[0]};
        for(int i = 1; i < nums.Length; i++)
        {
            if(nums[i] != n[n.Count-1])
            {
                n.Add(nums[i]);
            }
        }
        int hills = 0;
        for(int i = 1; i < n.Count-1; i++)
        {
            if(n[i]>n[i-1] && n[i]>n[i+1])
            {
                hills++;
            }
            else if(n[i]<n[i-1] && n[i]<n[i+1])
            {
                hills++;
            }
        }
        return hills;
    }
}